iothub/device/src/Transport/AmqpIoT/AmqpIoTResultAdapter.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Amqp.Framing;
+
+namespace Microsoft.Azure.Devices.Client.Transport.AmqpIoT
+{
+    internal enum AmqpIoTDisposeActions
+    {
+        Accepted,
+        Released,
+        Rejected
+    }
+
+    internal static class AmqpIoTResultAdapter
+    {
+        internal static Outcome GetResult(AmqpIoTDisposeActions amqpIoTConstants)
+        {
+            switch (amqpIoTConstants)
+            {
+                case AmqpIoTDisposeActions.Accepted:
+                    return new Accepted();
+                case AmqpIoTDisposeActions.Released:
+                    return new Released();
+                case AmqpIoTDisposeActions.Rejected:
+                    return new Rejected();
+                default:
```

Comment:
Be noted that this is letting _receivingAmqpLink.DisposeMessageAsync below decide how to dispose the message in case the value is null, which we then don't know how it does it (or it would if it changes anything).

I would say that it would be safer to throw an exception if the values cannot be mapped.

```
internal async Task<AmqpIoTOutcome> DisposeMessageAsync(string lockToken, Outcome outcome, TimeSpan timeout) 
        { 
            ArraySegment<byte> deliveryTag = ConvertToDeliveryTag(lockToken); 
            Outcome disposeOutcome =  
                await _receivingAmqpLink.DisposeMessageAsync( 
                    deliveryTag,  
                    outcome,  
```

Reply:
Fixed

---
In reply to: [293024536](https://github.com/Azure/azure-iot-sdk-csharp/pull/947#discussion_r293024536) [](ancestors = 293024536)